rfcs/006-package-management.md

```diff
@@ -0,0 +1,359 @@
+  system like Bazel. (However, providing integration points for other tools *is*
+  in scope. More detail on that below.)
+
+## The manifest file
+
+We will require a manifest file in order to import packages. Manifest files
+must be named `electroplate.ncl`, and they are found by searching up from the
+file being evaluated. That is, when the user invokes `nickel export path/to/
+foo.ncl`, we look for a manifest at `path/to/electroplate.ncl` and then at
+`path/electroplate.ncl`, and so on.
+
+The manifest file format is defined by the contract `std.package.Manifest`,
+which is defined as
+
+```nickel
+{
+  name | String,
+  version | Semver,
+  nickel-version | Semver,
+
+  dependencies
+    | { _:
+      [|
+        'Path String,
+        'Git { url | String, branch | optional | String, rev | optional | String },
+        'Index { name | String, version | Semver },
+      |]
+      }
+    | default
+    = {},
+}
+```
+
+So an example manifest might look like
+
+```nickel
+{
+  name = "demo"
+  version = "0.1.0"
+  nickel-version = "^1.0"
+  dependencies = {
+    foo = 'Index { package = "github/tweag/foo", version = "1.2.0" }
+    bar = 'Path "../my-bar",
+  }
+} | std.package.Manifest
+```
+
+### Alternative: inline dependencies
+
+Nix and Dhall allow for importing dependencies dynamically, using things like
+`fetchGit`. There was some discussion
+[here](https://github.com/tweag/nickel/issues/329#issuecomment-967372858)
+on the advantages and disadvantages of inline imports.
+
+### Alternative: toml manifest
+
+Maybe the manifest should be in some plain-data format like toml. This would
+be easier to modify programmatically, and it would prevent people from putting
+lots of complicated code in their manifest files.
+The disadvantage of a plain-data format is that you don't get a nice contract
+for it and you can't use ADTs for the dependencies.
+
+We discussed this point in office hours, and the general sentiment was that
+it's ok to allow the manifest to be interpreted. If someone wants to use
+that power to create a ridiculously complicated manifest, that's their problem.
+
+### Alternative: shorthand for registry imports
+
+Cargo allows a shorthand like
+
+```toml
+    "github/tweag/foo" = "1.2.0"
+```
+
+instead of
+
+```nickel
+    foo = 'Index { package = "github/tweag/foo", version = "1.2.0" }
+```
+
+Since we expect registry imports to be the common case, maybe it's worth having
+a shorthand?
+
+### Alternative: manifest file name
+
+Bikeshed the name "electroplate.ncl".
+
+## Import statements
+
+The manifest file assigns a name to each dependency; to import the dependency
+named `foo` you simply write `import foo`. That is, an `import` statement either
+takes a string in quotes -- in which case it imports a path -- or an identifier
+without quote -- in which case it imports a package.
+
+The `import foo` expression evaluates to the contents of `main.ncl` in `foo`'s
+root directory.
+
+### Alternative: other entry points
```

Comment:
> as long as imports aren't evaluated at all unless forced, because main.ncl may not exist

This might be an issue, because currently we do eagerly parse and typecheck imports.